### PR TITLE
Enhancement: Flow Editor Case validation

### PR DESCRIPTION
**Pull request by AkhileshNegi**
## Summary
 Target issue is #2481  

## Notes

- Added additional check to validate on wait-for-response case
- Updated testcase
- Fixed some typos

### CODE REVIEW

File: lib/glific/flows/case.ex [hunk 1]
```diff
@@ -83,8 +83,8 @@ defmodule Glific.Flows.Case do
   @doc """
   Validate a case
   """
-  @spec validate(Case.t(), Keyword.t(), map()) :: Keyword.t()
-  def validate(%{arguments: arguments} = _case, errors, flow) when arguments != [] do
+  @spec validate(Case.t(), Keyword.t(), map(), boolean()) :: Keyword.t()
+  def validate(%{arguments: arguments} = _case, errors, flow, false) when arguments != [] do
```
Comment (pankaj-ag) [inline lib/glific/flows/case.ex]: We need to apply the validation if has_wait is true right? not false

File: lib/glific/flows/router.ex [hunk 2]
```diff
@@ -156,7 +156,7 @@ defmodule Glific.Flows.Router do
       router.cases
       |> Enum.reduce(
         errors,
-        &Case.validate(&1, &2, flow)
+        &Case.validate(&1, &2, flow, is_nil(router.wait))
```
Comment (pankaj-ag) [inline lib/glific/flows/router.ex]: Semantically this is not correct. Either send router.wait as a true value or fix the Case.validate cluase.

**Review by pankaj-ag — CHANGES_REQUESTED** [review]

File: lib/glific/flows/router.ex (hunk 2 above)
Comment (AkhileshNegi) [reply lib/glific/flows/router.ex]: @pankaj-ag made requested changes

File: lib/glific/flows/case.ex (hunk 1 above)
Comment (AkhileshNegi) [reply lib/glific/flows/case.ex]: @pankaj-ag made requested changes

**Review by pankaj-ag — APPROVED** [review]

**Pull request merged**